cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -507,6 +507,41 @@ TEST_F(AzuriteFileSystemTest, CreateDirUri) {
   ASSERT_RAISES(Invalid, fs_->CreateDir("abfs://" + RandomContainerName(), true));
 }
 
+TEST_F(AzuriteFileSystemTest, DeleteContainerDirSuccess) {
+  auto container_name = RandomContainerName();
+  ASSERT_OK(fs_->CreateDir(container_name));
+  ASSERT_OK(fs_->DeleteDir(container_name));
```

Comment:
Should we add a `arrow::fs::AssertFileInfo(fs_.get(), path, FileType::NotFound);` at the end? Personally I would probably also add an assertion that the container does exist before deleting it.

Reply:
OK. I'll add them.